app/models/in_progress_form.rb

```diff
@@ -1,8 +1,16 @@
 # frozen_string_literal: true
 class InProgressForm < ActiveRecord::Base
   attr_encrypted :form_data, key: Settings.db_encryption_key
+  validates(:form_data, presence: true)
+  before_save :serialize_form_data
 
   def self.form_for_user(form_id, user)
     InProgressForm.find_by(form_id: form_id, user_uuid: user.uuid)
   end
+
+  private
+
+  def serialize_form_data
+    self.form_data = form_data.to_json unless form_data.is_a?(String)
```

Comment:
Mmm...it might be useful to be able to query on metadata. For example, to see how many people stop on a specific page, or to see what the oldest form version is. To be fair, I don't know _why_ that would be useful, but I'd rather play it safe just in case it is. `metadata` won't contain any sensitive data needing encrypted, at any rate.